daml-lf/archive/src/main/protobuf/com/digitalasset/daml_lf_dev/daml_lf_1.proto

```diff
@@ -112,6 +112,21 @@ message TypeConName {
   }
 }
 
+// A fully qualified reference to a type synonym name.
+message TypeSynName {
+
+  // Module where the type is defined.
+  ModuleRef module = 1;
+
+  oneof name {
```

Comment:
We enforce that everything is interned for DAML-LF `>= 1.7` so we don’t need to have a non-interned version of this.